tests/ui/missing-trait-bounds/issue-35677.fixed

```diff
@@ -3,7 +3,7 @@
 use std::collections::HashSet;
 use std::hash::Hash;
 
-fn is_subset<T>(this: &HashSet<T>, other: &HashSet<T>) -> bool where T: Eq, T: Hash {
+fn is_subset<T>(this: &HashSet<T>, other: &HashSet<T>) -> bool where T: Eq, T: Hash, T: PartialEq {
```

Comment:
Hm, I don't consider this an improvement -- `T: Eq` implies `T: PartialEq` by elaboration.

Reply:
I agree, so I tested hiding the suggestions when a parent obligation was present, but that broke some existing tests with applicable suggestions. I'll see if we can inspect the obligation cause code instead to avoid these.

Reply:
Fixed